app/Repositories/ModelRepositories/AnalysisRepositories/ModelChangeRepository.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+namespace App\Entity\Repositories;
+
+use App\Entity\ModelChange;
+use App\Entity\ModelReaction;
+use App\Entity\ModelSpecie;
+use App\Entity\ModelReactionItem;
+use App\Entity\IdentifiedObject;
+use App\Entity\ModelTask;
+use Doctrine\ORM\EntityManager;
+use Doctrine\ORM\QueryBuilder;
+
+
+class ModelChangeRepository implements IDependentSBaseRepository
+{
+    /** @var EntityManager * */
+    protected $em;
+
+    /** @var \Doctrine\ORM\ModelChangeRepository */
+    private $repository;
+
+    /** @var IdentifiedObject */
+    private $parent;
+
+    public function __construct(EntityManager $em)
+    {
+        $this->em = $em;
+        $this->repository = $em->getRepository(ModelChange::class);
+    }
+
+    protected static function getParentClassName(): array
+    {
+        return [ModelTask::class];
+    }
+
+    public function getParent(): IdentifiedObject
+    {
+        return $this->parent;
+    }
+
+    public function get(int $id)
+    {
+        return $this->em->find(ModelChange::class, $id);
+    }
+
+    public function getNumResults(array $filter): int
+    {
+        return ((int)$this->buildListQuery($filter)
+            ->select('COUNT(c)')
+            ->getQuery()
+            ->getScalarResult());
+    }
+
+    public function getList(array $filter, array $sort, array $limit): array
+    {
+        $query = $this->buildListQuery($filter)
+            ->select('c.id, c.task_id, c.type, c.origin_id, c.value');
+
+        return $query->getQuery()->getArrayResult();
+    }
+
+    public function setParent(IdentifiedObject $object): void
+    {
+        dump($object);
```

Comment:
clean up 🗑 (searching the project for dumps and exits before commiting is not a bad idea)